src/components/ModelingMachineProvider.tsx

```diff
@@ -258,20 +258,18 @@ export const ModelingMachineProvider = ({
         ),
         // For some reason, this one is tricky.
         'Get selection from ast and sketchDetails': assign(
-          // @ts-ignore
-          ({ sketchDetails, selectionRanges }, event) => {
-            // @ts-ignore
-            event.data = {
-              // @ts-ignore
+          ({ sketchDetails, selectionRanges }) => {
+            const updatedSelections = updateSelections(
+              [sketchDetails?.sketchPathToNode || []],
+              selectionRanges,
+              kclManager.ast
```

Comment:
does this break anthing? passing event.data like this was to fix a bug

Reply:
Okay! thanks for prompting me a bit more on this, so if. . . 

https://github.com/KittyCAD/modeling-app/assets/29681384/024d9b28-d11c-4be4-b25a-9a458cfaa5d0

If that's right then it's been fixed but without the hackiness,

So now I understand that what you were doing with the `Get selection from ast and sketchDetails` action was mutating the data in flight just before the `Set selection` action was called. Clever but absolutely not a precedent we want in Xstate, and is why or that typescript hacks were needed, I've since changed the invoke service `Get convert to variable info` to confirm to to the rest of the `get ... info` invoked services (these guys 👇 )

![image](https://github.com/KittyCAD/modeling-app/assets/29681384/64a5ecc1-2e97-4f49-a9a1-3c2d3aa57a09)

So that only `Set selection` action is needed.

Does it look like this incorporates your fix, but without the magic?

Reply:
That sounds perfect then. Yeah I'm not very familiar with XState and I was fighting it a lot so your code here is very helpful to learn from - thank you!